pandas/core/groupby.py

```diff
@@ -2883,7 +2883,12 @@ def aggregate(self, func_or_funcs, *args, **kwargs):
             except Exception:
                 result = self._aggregate_named(func_or_funcs, *args, **kwargs)
 
-            index = Index(sorted(result), name=self.grouper.names[0])
+            # mixed types fail to sort
+            try:
+                values = sorted(result)
+            except TypeError:
+                values = result
```

Comment:
I tried this change and it caused 4 tests in the full test suite to fail. Those tests were TestDatetimeIndex.test_resample_categorical_data_with_timedeltaindex, TestGroupByAggregate.test_mixed_type_grouping (the test that I wrote), TestGroupBy.test_basic, and test_compressed_urls[python-explicit-gzip-.gz]
Also, I noticed that safe_sort returns all objects as strings, even if the input includes numeric types.